src/js/components/Data/stories/SpaceX.js

```diff
@@ -131,38 +132,43 @@ const columns = [
   },
 ];
 
-export const Table = () => {
-  const [data, setData] = useState();
-  const [rockets, setRockets] = useState();
-  const [view, setView] = useState({ search: '' });
-  const [sort, setSort] = useState({ property: 'name', direction: 'asc' });
-  const [page, setPage] = useState(1);
-  const limit = 10;
-
-  const search = view.search || '';
+export const SpaceX = () => {
+  const [total, setTotal] = useState(0);
+  const [result, setResult] = useState({ data: [] });
+  const [rockets, setRockets] = useState([]);
+  const [view, setView] = useState({
+    search: '',
+    sort: { property: 'name', direction: 'asc' },
+    step: 10,
+  });
 
   useEffect(() => {
-    fetchRockets().then((d) => setRockets(d));
+    fetchRockets().then((response) =>
+      setRockets(
+        response.docs.map(({ name, id }) => ({ value: id, label: name })),
+      ),
+    );
   }, []);
 
   useEffect(() => {
-    fetchLaunches({
-      search,
-      limit,
-      page,
-      sort,
-      properties: view.properties,
-    }).then((d) => setData(d));
-  }, [search, limit, page, sort, view.properties]);
-
-  const numberItems = data?.totalDocs || 0;
-  const pageResultStart = (page - 1) * limit + 1;
-  const pageResultEnd = Math.min(page * limit, numberItems);
-
-  const rocketOptions =
-    rockets?.docs?.map(({ name, id }) => ({ value: id, label: name })) || [];
-
-  const docs = data?.docs || [];
+    fetchLaunches(view).then((response) => {
+      setResult({
+        data: response.docs,
+        filteredTotal: response.totalDocs,
+        page: response.page,
+      });
+      setTotal((prevTotal) => Math.max(prevTotal, response.totalDocs));
```

Comment:
I'm not really sure why this is doing the max with the previous.

Reply:
It appeared to me that you don't get the total number of items that are available in a request that has filtered some out. So, if your first request is for all Falcon 1 launches, you don't get anything in the response indicating that there are 205 total launches even though there are only 5 Falcon 1 launches. The Math.max() is a weak safety net in case the first request has a filter that doesn't get everything. I guess it would be more accurate to detect this case and do a subsequent query with a small limit just to get the total.

Reply:
I've added a comment to explain.